### DISCUSSION TITLE
Redis as a cache

### DISCUSSION THREAD

**Discussion by virus2016** [directus/directus#4742] [Archive]
Hi there,

I have added redis as a cache however I am not seeing improvments to the latency. It takes around 200ms to download an asset with cache enabled. My understanding is if the cache has already accessed the asset from a S3 bucket that it will cache that response and serve it when someone else asks for it. 

Is there debuging I can do to get to the route of the issue?

This could be related to #4740?

**Comment by rijkvanzanten** [comment]
The assets aren't cached in Redis, so that would be accurate. Redis is used to cache the JSON output for any given request. Assets are generated first time they're requested, and then saved on your storage adapter. Subsequent reads stream it straight from the pre-generated asset on your storage adapter

**Reply by virus2016** [reply]
Are we able to read from an s3 bucket then store locally for faster reads?

**Reply by rijkvanzanten** [reply]
Directus doesn't do that natively, but you should be able to create a custom hook to do that 👍🏻
